File: HRMS/HRMS.Web/Modules/ServerTypes/HR/DesignationColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DesignationRow } from "./DesignationRow";

export interface DesignationColumns {
    DesignationId: Column<DesignationRow>;
    DesignationName: Column<DesignationRow>;
    Level: Column<DesignationRow>;
    IsActive: Column<DesignationRow>;
}

export class DesignationColumns extends ColumnsBase<DesignationRow> {
    static readonly columnsKey = 'HR.Designation';
    static readonly Fields = fieldsProxy<DesignationColumns>();
}

File: HRMS/HRMS.Web/Modules/HR/Designation/DesignationColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.HR.Columns;

[ColumnsScript("HR.Designation")]
[BasedOnRow(typeof(DesignationRow), CheckNames = true)]
public class DesignationColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DesignationId { get; set; }
    [EditLink]
    public string DesignationName { get; set; }
    public int Level { get; set; }
    [Width(100), AlignCenter]
    public bool IsActive { get; set; }
}
